File: digital-assignment-1/contrast_adjustment.m
% program for contrast adjustment
clc 
clear
close all

% reading the images
A = imread('images\picture1.jpg');

% contrast adjustment
B = A*1.5;

% display input and output images
subplot(1, 2, 1);
imshow(A);
title('Original Image');
subplot(1, 2, 2);
imshow(B);
title('Image after Contrast Adjustment');

% metric to compare the results
nA =niqe(A);
nB = niqe(B);
fprintf("Image score for orginal image: %0.2f.\n", nA)
fprintf("Image score for contrast-adjusted image: %0.2f.\n", nB)

File: digital-assignment-2/contrast_adjustment.py
import cv2

# Load the image
img = cv2.imread('images/picture1.jpg', 0)

# Define the parameters for contrast adjustment
alpha = 1.5 # Contrast control (1.0-3.0)
beta = 0 # Brightness control (0-100)

# Apply contrast adjustment
img_contrast = cv2.convertScaleAbs(img, alpha=alpha, beta=beta)

# Display the results
cv2.imshow('Input Image', img)
cv2.imshow('Contrast Adjusted Image', img_contrast)
cv2.waitKey(0)
cv2.destroyAllWindows()
